pkg/controllers/nodeclass/controller.go

```diff
@@ -84,18 +80,23 @@ func NewController(
 	capacityReservationProvider capacityreservation.Provider,
 	ec2api sdk.EC2API,
 ) *Controller {
-
+	reconcilers := []nodeClassReconciler{
+		NewAMIReconciler(amiProvider),
+		&Subnet{subnetProvider: subnetProvider},
+		&SecurityGroup{securityGroupProvider: securityGroupProvider},
+		&InstanceProfile{instanceProfileProvider: instanceProfileProvider},
+		&Validation{ec2api: ec2api, amiProvider: amiProvider},
+		&Readiness{launchTemplateProvider: launchTemplateProvider},
```

Comment:
Is readiness even the right name for this anymore? I get this is unrelated to this PR but I can't even remember what this check is doing anymore? Should this be in validation?

Reply:
Maybe? The only check at the moment is to ensure we can resolve the cluster CIDR for AL2023. I would need to revisit the comments from that PR, but I believe the rationale to drive it through the ready condition rather than adding a standalone condition was that we only ever need to resolve the cluster CIDR for NodeClasses with AL2023 - otherwise it was perfectly fine to be unable to resolve it.

Reply:
Either way, I think we can consider this out-of-scope of this PR.

Reply:
Let's take a note and address it in another PR